packages/ozone/src/api/moderation/emitEvent.ts

```diff
@@ -162,6 +162,10 @@ const handleModerationEvent = async ({
       }
     }
 
+    if (isTakedownEvent && result.event.meta?.acknowledgeAccountSubjects) {
+      await moderationTxn.resolveSubjectsForAccount(subject.did, createdBy)
+    }
```

Comment:
My biggest question about this work is: should the flag be tied specifically to the takedown event?  It becomes a bit of a complicated idea, feels like the API reflects a fairly specific workflow, and we could probably peel these ideas apart without sacrificing much.  As a bulk action, the failure modes of the acknowledgement are also pretty different from the takedown.

Since this deals with acknowledgement, I'd propose to instead add a flag to `#modEventAcknowledge`, e.g. something like `event.applyToAllRepoSubjects`.  And then the UI would perform this bulk acknowledgement after successfully performing the takedown.

Any thoughts/feels?

Reply:
I can totally see how adding the flag to ack event is useful and helps us decouple this from a particular event and re-use in other cases. However, I am not sure if we have such cases where we'd want to resolve all subjects without review outside of an account level takedown?

there's also this concern around surfacing these in "mod history" and while we should be able to get away with unreviewed "ack" on records if the author account is takendown but doing this in other cases might raise questions to the viewer around "why reports were acked without action on the subject".

I don't feel too strongly about either direction tbh but I like the restriction that only takedown allows us to do this seemingly aggressive ack without review.